src/Drupal/Driver/Fields/Drupal7/AbstractHandler.php

```diff
@@ -22,8 +26,8 @@
    * @param $field_name
    * @return mixed
    */
-  public function __construct($entity_type, $field_name) {
+  public function __construct($entity_type, $field_name, $language) {
     $this->field_info = field_info_field($field_name);
+    $this->language = field_is_translatable('node', $this->field_info) ? $language : LANGUAGE_NONE;
```

Comment:
We could actually use something similar to what core does for comments:

```php
$langcode = field_is_translatable('comment', $field) ? entity_language('comment', $comment) : LANGUAGE_NONE;
```

So:

```php
$langcode = field_is_translatable($entity_type, $this->field_info) ? entity_language($entity_type, $entity) : LANGUAGE_NONE;
```